File: xflaim/src/cs/wrapper/xflaim/DbCopyStatus.cs
//------------------------------------------------------------------------------
// Desc:	Db Copy Status
// Tabs:	3
//
// Copyright (c) 2006-2007 Novell, Inc. All Rights Reserved.
//
// This library is free software; you can redistribute it and/or
// modify it under the terms of the GNU Lesser General Public
// License as published by the Free Software Foundation; version 2.1
// of the License.
//
// This library is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Library Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public
// License along with this library; if not, contact Novell, Inc.
//
// To contact Novell about this file by physical or electronic mail, 
// you may find current contact information at www.novell.com.
//
// $Id$
//------------------------------------------------------------------------------

using System;
using System.Runtime.InteropServices;

namespace xflaim
{

	/// <summary>
	/// This interface allows XFlaim to periodically pass information back to the
	/// client about the status of an ongoing database copy operation.  The
	/// implementor may do anything it wants with the information, such as write
	/// it to a log file or display it on the screen.
	/// </summary>
	public interface DbCopyStatus 
	{

		/// <summary>
		/// The <see cref="DbSystem.dbCopy"/> operation calls this method periodically to
		/// inform the implementation object about the status of the copy operation.
		/// </summary>
		/// <param name="ulBytesToCopy">
		/// The total number of bytes that the <see cref="DbSystem.dbCopy"/> operation
		/// will ultimately copy.
		/// </param>
		/// <param name="ulBytesCopied">
		/// The number of bytes that the <see cref="DbSystem.dbCopy"/> operation has
		/// copied so far.
		/// </param>
		/// <param name="sSrcFileName">
		/// The name of the file that is currently being copied.  This is only passed
		/// when the source file name is changed.  Otherwise, it will be null.
		/// </param>
		/// <param name="sDestFileName">
		/// The name of the current destination file.  This is only passed when the
		/// destination file name is changed.  Otherwise, it will be null.
		/// </param>
		/// <returns>
		/// If the implementation object returns anything but RCODE.NE_XFLM_OK
		/// the <see cref="DbSystem.dbCopy"/> operation will abort and throw an
		/// <see cref="XFlaimException"/>
		/// </returns>
		RCODE dbCopyStatus(
			ulong			ulBytesToCopy,
			ulong			ulBytesCopied,
			string		sSrcFileName,
			string		sDestFileName);
	}
}

File: xflaim/src/java/wrapper/xflaim/DbCopyStatus.java
//------------------------------------------------------------------------------
// Desc:	Db Copy Status
// Tabs:	3
//
// This library is free software; you can redistribute it and/or
// modify it under the terms of the GNU Lesser General Public
// License as published by the Free Software Foundation; version 2.1
// of the License.
//
// This library is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Library Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public
// License along with this library; if not, contact Novell, Inc.
//
// To contact Novell about this file by physical or electronic mail, 
// you may find current contact information at www.novell.com.
//
// $Id$
//------------------------------------------------------------------------------

package xflaim;

/**
 * This interface alows XFlaim to periodically pass information back to the
 * client about the status of an ongoing database copy operation.  The
 * implementor may do anything it wants with the information, such as write
 * it to a log file or display it on the screen.
 */
public interface DbCopyStatus 
{
	/**
	 * Called periodically to inform the client about the status of the copy
	 * operation.
	 * @param ui64BytesToCopy The total number of bytes that this operation
	 * will copy.
	 * @param ui64BytesCopied The number of bytes that have been copied so far.
	 * @param bNewSrcFile Set to true if the copy operation has started
	 * working on a new file. 
	 * @param pszSrcFileName The name of the file that is currently being copied.
	 * @param pszDestFileName The name of the destination file.
	 * @return Returns a status code.  The integer should one of the constants
	 * found in {@link xflaim.RCODE xflaim.RCODE}.
	 * Note that returning anything other than NE_XFLM_OK will cause the
	 * copy operation to abort and an XFLaimException to be thrown.
	 */
	int dbCopyStatus(
		long			ui64BytesToCopy,
		long			ui64BytesCopied,
		boolean		bNewSrcFile,
		String		pszSrcFileName,
		String		pszDestFileName);
}
